Source/Core/DolphinQt/Debugger/MemoryWidget.cpp

```diff
@@ -616,46 +584,40 @@ void MemoryWidget::OnDumpFakeVMEM()
             std::distance(accessors->begin(), accessors->end()));
 }
 
-std::vector<u8> MemoryWidget::GetValueData() const
+bool MemoryWidget::IsValueValid() const
 {
-  std::vector<u8> search_for;  // Series of bytes we want to look for
-
   if (m_find_ascii->isChecked())
-  {
-    const QByteArray bytes = m_data_edit->text().toUtf8();
-    search_for.assign(bytes.begin(), bytes.end());
-  }
-  else
-  {
-    bool good;
-    u64 value = m_data_edit->text().toULongLong(&good, 16);
-
-    if (!good)
-      return {};
+    return true;
+  const QRegularExpression is_hex(QStringLiteral("^([0-9A-F]{2})*$"),
```

Comment:
Yes, that's intended for two reasons:
 - To make sure that the value is a valid hex string that `QByteArray::fromHex` will parse properly.
 - To enforce the fact it accepts hex string and not an integer (the previous behaviour was confusing and limited to a 64-bit integer).

Reply:
I agree the value should (and does in your PR) allow values longer than 64 bits.

The connection between "only accepts hex pairs" and "accepts arbitrary length hex strings instead of 64-bit integers" is fuzzy to me. If we changed the Value placeholder in hex mode to "Hex string" or something along those lines I feel that would be more explicit while still allowing people to type 0 instead of 00.

To make sure the 0x prefix doesn't interfere with ::fromHex we could match `"^(?:0x)?([0-9A-F]+)$"` and use capture group 1 to get the non-prefixed string (the implicitly-generated group 0 being the full match).

Reply:
What about changing the "Hex" radio button's label to "Hex string" instead to make it clearer?

Reply:
That sounds good to me.